mongodbatlas/resource_mongodbatlas_data_lake_test.go

```diff
@@ -0,0 +1,172 @@
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMongoDBAtlasDataLakeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasDataLakeConfig(projectName, orgID, name, roleID, testS3Bucket),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+				),
+			},
+			{
+				Config: testAccMongoDBAtlasDataLakeConfig(projectName, orgID, name, roleID, testS3BucketUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceMongoDBAtlasDataLake_importBasic(t *testing.T) {
+	SkipTestExtCred(t)
+	var (
+		resourceName = "mongodbatlas_data_lake.basic_ds"
+		orgID        = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName  = acctest.RandomWithPrefix("test-acc")
+		name         = acctest.RandomWithPrefix("test-acc")
+		roleID       = os.Getenv("AWS_ROLE_ID")
+		testS3Bucket = os.Getenv("AWS_S3_BUCKET")
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMongoDBAtlasDataLakeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasDataLakeConfig(projectName, orgID, name, roleID, testS3Bucket),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccCheckMongoDBAtlasDataLakeImportStateIDFunc(resourceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckMongoDBAtlasDataLakeImportStateIDFunc(resourceName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return "", fmt.Errorf("not found: %s", resourceName)
+		}
+
+		ids := decodeStateID(rs.Primary.ID)
+
+		return fmt.Sprintf("%s-%s", ids["project_id"], ids["name"]), nil
+	}
+}
+
+func testAccCheckMongoDBAtlasDataLakeExists(resourceName string, dbUser *matlas.DataLake) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*matlas.Client)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.Attributes["project_id"] == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		ids := decodeStateID(rs.Primary.ID)
+
+		if dbUserResp, _, err := conn.DataLakes.Get(context.Background(), ids["project_id"], ids["name"]); err == nil {
+			*dbUser = *dbUserResp
```

Comment:
I'm not sure what the dbUser is from or for - might just be a confusing variable name as I'm not sure where database user is coming from?